code/common/src/test/scala/com/avast/cloud/datadog4s/inmemory/InMemoryMetricFactory.scala

```diff
@@ -0,0 +1,63 @@
+package com.avast.cloud.datadog4s.inmemory
+
+import java.time.Duration
+import java.util.concurrent.{ ConcurrentHashMap, ConcurrentMap }
+
+import cats.effect.Sync
+import cats.syntax.functor._
+import cats.syntax.flatMap._
+import com.avast.datadog4s.api.metric._
+import com.avast.datadog4s.api.{ GaugeFactory, HistogramFactory, MetricFactory, Tag }
+
+class InMemoryMetricFactory[F[_]: Sync](val state: ConcurrentMap[String, Vector[Record[Any]]])
```

Comment:
Why don't you use `Ref` instead?

Reply:
hmmm i thought there would be issues with that but on second thought it should be possible